engine/table/src/test/java/io/deephaven/engine/table/impl/updateby/TestRollingMinMax.java

```diff
@@ -0,0 +1,1188 @@
+
+        doTestAppendOnly(true, prevTicks, postTicks);
+    }
+
+    @Test
+    public void testZeroKeyAppendOnlyTimedRev() {
+        final Duration prevTime = Duration.ofMinutes(10);
+        final Duration postTime = Duration.ofMinutes(0);
+
+        doTestAppendOnlyTimed(false, prevTime, postTime);
+    }
+
+    @Test
+    public void testZeroKeyAppendOnlyTimedRevExclusive() {
+        final Duration prevTime = Duration.ofMinutes(10);
+        final Duration postTime = Duration.ofMinutes(-5);
+
+        doTestAppendOnlyTimed(false, prevTime, postTime);
+    }
+
+    @Test
+    public void testZeroKeyAppendOnlyTimedFwd() {
+        final Duration prevTime = Duration.ofMinutes(0);
+        final Duration postTime = Duration.ofMinutes(10);
+
+        doTestAppendOnlyTimed(false, prevTime, postTime);
+    }
+
+    @Test
+    public void testZeroKeyAppendOnlyTimedFwdExclusive() {
+        final Duration prevTime = Duration.ofMinutes(-5);
+        final Duration postTime = Duration.ofMinutes(10);
+
+        doTestAppendOnlyTimed(false, prevTime, postTime);
+    }
+
+    @Test
+    public void testZeroKeyAppendOnlyTimedFwdRev() {
+        final Duration prevTime = Duration.ofMinutes(5);
+        final Duration postTime = Duration.ofMinutes(5);
+
+        doTestAppendOnlyTimed(false, prevTime, postTime);
+    }
+
+    @Test
+    public void testBucketedAppendOnlyTimedRev() {
+        final Duration prevTime = Duration.ofMinutes(10);
+        final Duration postTime = Duration.ofMinutes(0);
+
+        doTestAppendOnlyTimed(true, prevTime, postTime);
+    }
+
+    @Test
+    public void testBucketedAppendOnlyTimedRevExclusive() {
+        final Duration prevTime = Duration.ofMinutes(10);
+        final Duration postTime = Duration.ofMinutes(-5);
+
+        doTestAppendOnlyTimed(true, prevTime, postTime);
+    }
+
+    @Test
+    public void testBucketedAppendOnlyTimedFwd() {
+        final Duration prevTime = Duration.ofMinutes(0);
+        final Duration postTime = Duration.ofMinutes(10);
+
+        doTestAppendOnlyTimed(true, prevTime, postTime);
+    }
+
+    @Test
+    public void testBucketedAppendOnlyTimedFwdExclusive() {
+        final Duration prevTime = Duration.ofMinutes(-5);
+        final Duration postTime = Duration.ofMinutes(10);
+
+        doTestAppendOnlyTimed(true, prevTime, postTime);
+    }
+
+    @Test
+    public void testBucketedAppendOnlyTimedFwdRev() {
+        final Duration prevTime = Duration.ofMinutes(5);
+        final Duration postTime = Duration.ofMinutes(5);
+
+        doTestAppendOnlyTimed(true, prevTime, postTime);
+    }
+
+    private void doTestAppendOnly(boolean bucketed, int prevTicks, int postTicks) {
+        final CreateResult result = createTestTable(DYNAMIC_TABLE_SIZE, bucketed, false, true, 0x31313131);
+        final QueryTable t = result.t;
+        t.setAttribute(Table.ADD_ONLY_TABLE_ATTRIBUTE, Boolean.TRUE);
+
+        final EvalNugget[] nuggets = new EvalNugget[] {
+                new EvalNugget() {
```

Comment:
Similar comment here (as on other PR); I do like the compaction that comes from using `EvalNugget.from(Supplier<Table> makeTable)` rather than subclassing && implementing `Table e()` directly.